### PR TITLE
React react1 week3

### PR DESCRIPTION
**Pull request by danJecu**
Completed tasks for react-react1-week3 homework.

### CODE REVIEW

File: src/client/components/MealsList.jsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      } else {
+        res = await fetch('/api/meals');
+      }
+
+      if (res.ok) {
+        const json = await res.json();
+        dispatch({ type: 'SET_MEALS', payload: json });
+      }
+    })();
+  }, [dispatch, searchQuery]);
```
Comment (Divyajg) [inline src/client/components/MealsList.jsx:21]: When you use React Context Concept, Above Fetch should go into the Context.

File: src/client/contexts/MealsContext.jsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  const mealsContextValue = useMemo(
+    () => ({ ...state, dispatch, handleModalOpen }),
+    [state]
+  );
+
+  return (
+    <MealsContext.Provider value={mealsContextValue}>
+      {children}
+    </MealsContext.Provider>
+  );
```
Comment (Divyajg) [inline src/client/contexts/MealsContext.jsx:56]: Have not provided the children with the context anywhere or Did I miss seeing it?

**Review by Divyajg — COMMENTED** [review]
I haven't run your code, But I left a couple of comments. Let me know If you need any more info!
Get to me in Slack if you have any questions.

File: src/client/contexts/MealsContext.jsx (hunk 2 above)
Comment (danJecu) [reply src/client/contexts/MealsContext.jsx:56]: I forgot to wrap the App.
It's done.

File: src/client/components/MealsList.jsx (hunk 1 above)
Comment (danJecu) [reply src/client/components/MealsList.jsx]: I was working on my search bar and I think I left the code there.
I removed for this PR.

**Pull request merged**